src/datasources/provider-datasource.ts

```diff
@@ -0,0 +1,27 @@
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { BaseOpenMRSDataSource } from './data-source';
+
+export class ProviderDataSource extends BaseOpenMRSDataSource {
+  constructor() {
+    super(`${restBaseUrl}/provider?v=custom:(uuid,display)`);
+  }
+
+  fetchData(searchTerm: string, config?: Record<string, any>, uuid?: string): Promise<any[]> {
+    let apiUrl = this.url;
+    const urlParts = apiUrl.split('?');
+    if (config?.tag) {
+      apiUrl = `${urlParts[0]}?tag=${config.tag}&${urlParts[1]}`;
+    }
+    //overwrite url if there's a uuid value, meaning we are in edit mode
+    if (uuid) {
+      apiUrl = `${urlParts[0]}/${uuid}?${urlParts[1]}`;
+    }
+
+    return openmrsFetch(searchTerm ? `${apiUrl}&q=${searchTerm}` : apiUrl).then(({ data }) => {
+      if (data.results) {
+        return data.results;
+      }
+      return data;
+    });
+  }
```

Comment:
This implementation is unnecessarily complicated likely because of a poor abstraction in the `BaseOpenMRSDataSource`. I also see copy and paste residue eg. the part where you're applying a "tag"; providers don't support Tags. I would refactor to simply:

```ts
export class ProviderDataSource extends BaseOpenMRSDataSource {
  constructor() {
    super(null);
  }

  async fetchData(searchTerm: string, config?: Record<string, any>, uuid?: string): Promise<any[]> {
    const rep = 'v=custom:(uuid,display)';
    const url = uuid ? `${restBaseUrl}/provider/${uuid}?${rep}` : `${restBaseUrl}/provider?${rep}`;
    const { data } = await openmrsFetch(searchTerm ? `${url}&q=${searchTerm}` : url);
    if (data.results) {
      return data.results;
    }
    return Array.isArray(data) ? data : [data];
  }
}
```